File: API/APIMeow/Models/Usuario.cs

using System.ComponentModel.DataAnnotations;

public class Usuario
{
    [Key]
    public int IdUsuario { get; set; }
    public string Senha { get; set; }
    public string Biografia { get; set; }
    public string Nome { get; set; }
    public string Email { get; set; }
    public decimal Saldo { get; set; }
    public int Pontos { get; set; }
}

File: front_meow/lib/model/usuario.dart
class Usuario {
  final int idUsuario;
  final String nome;
  final String email;
  final double saldo;
  final int pontos;
  final String biografia;

  Usuario({
    required this.idUsuario,
    required this.nome,
    required this.email,
    required this.saldo,
    required this.pontos,
    required this.biografia,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return switch (json) {
      {
        'idUsuario': int idUsuario,
        'nome': String nome,
        'email': String email,
        'saldo': double saldo,
        'pontos': int pontos,
        'biografia': String biografia,
      } =>
        Usuario(
          idUsuario: idUsuario,
          nome: nome,
          email: email,
          saldo: saldo,
          pontos: pontos,
          biografia: biografia,
        ),
      _ => throw Exception('Erro ao carregar usuario'),
    };
  }
}
